src/libs/antares/study/CMakeLists.txt

```diff
@@ -0,0 +1,447 @@
+        parts/load/series.cpp
+        parts/load/prepro.cpp
+        parts/load/prepro.h
+        parts/load/prepro.hxx
+
+        # Solar
+        parts/solar/container.h
+        parts/solar/container.cpp
+        parts/solar/series.h
+        parts/solar/series.hxx
+        parts/solar/series.cpp
+        parts/solar/prepro.cpp
+        parts/solar/prepro.h
+        parts/solar/prepro.hxx
+
+        # Study
+        header.h
+        header.hxx
+        header.cpp
+        load-options.h
+        load-options.cpp
+        runtime/runtime.h
+        runtime/runtime.hxx
+        runtime/runtime.cpp
+        runtime.h
+        study.h
+        study.hxx
+        fwd.h
+        fwd.cpp
+        study.h
+        study.cpp
+        correlation-updater.hxx
+        study.importprepro.cpp
+        importlogs.cpp
+        load.cpp
+        save.cpp
+        study.extra.cpp
+        version.h
+        version.cpp
+        cleaner.h
+        cleaner/cleaner.cpp cleaner/versions.h
+        cleaner/cleaner-v20.cpp
+        output.cpp
+        output.h
+        output.hxx
+        layerdata.cpp
+        layerdata.h
+
+        # progression
+        progression/progression.h
+        progression/progression.hxx
+        progression/progression.cpp
+
+        # Simulation
+        simulation.h
+        simulation.cpp
+
+        # Filter
+        filter.h
+        filter.cpp
+
+        # Sets
+        sets.h
+        sets.hxx
+
+        # variable selection
+        variable-print-info.h
+        variable-print-info.cpp
+)
+
+source_group("study\\study" FILES ${SRC_STUDY})
+
+set(SRC_ALL
+        ${SRC_STDCXX}
+        ${SRC_AREAS}
+        ${SRC_XCAST}
+        ${SRC_STUDY_AREA_UI}
+        ${SRC_STUDY_ACTIONS}
+        ${SRC_STUDY_ACTIONS_HANDLER_ANTARES_STUDY_THERMAL}
+        ${SRC_STUDY_ACTIONS_HANDLER_ANTARES_STUDY_AREA}
+        ${SRC_STUDY_ACTIONS_HANDLER_ANTARES_STUDY_LINK}
+        ${SRC_STUDY_ACTIONS_HANDLER_ANTARES_STUDY_CONSTRAINT}
+        ${SRC_STUDY_UI}
+        ${SRC_STUDY_PARAMETERS}
+        ${SRC_STUDY_FINDER}
+        ${SRC_STUDY_BINDING_CONSTRAINT}
+        ${SRC_STUDY_PART_WIND}
+        ${SRC_STUDY_PART_HYDRO}
+        ${SRC_STUDY_PART_COMMON}
+        ${SRC_STUDY_PART_THERMAL}
+        ${SRC_STUDY_PART_RENEWABLE}
+        ${SRC_STUDY_PART_SHORT_TERM_STORAGE}
+        ${SRC_STUDY_SCENARIO_BUILDER}
+        ${SRC_STUDY}
+)
+
+add_library(study ${SRC_ALL})
+add_library(Antares::study ALIAS study)
+
+target_link_libraries(study
+    PUBLIC
+        yuni-static-core
+        Boost::boost
+        Antares::array #xcast
+        Antares::date #parameters
+        Antares::inifile #parameters
+        Antares::stdcxx #enum.hxx
+        Antares::utils #area.hxx
+        Antares::correlation
+        Antares::path
+        Antares::mersenne
+        Antares::result_writer #study.h
+        Antares::object
+    PRIVATE
+        Antares::exception
+)
+
+target_include_directories(study
+    PUBLIC
+        ${CMAKE_CURRENT_SOURCE_DIR}/../..
```

Comment:
IMO this should avoided just like `#include "../../header.hpp"`

Reply:
I've added some comment
#Dirty trick to allow inclusion via <antares/study/X>
# Make more than just study visible but it's the lesser evil for now

The proper way would be to have a proper include folder with all study includes (or split study in several library/objects) but it is in itself a task